### PR TITLE
[Quantization][Feature] Add AWQ quantization in vllm-ascend.

### PR DESCRIPTION
**Pull request by menogrey**
### What this PR does / why we need it?
Add AWQ quantization in vllm-ascend. Most of the code refer to [existing implement](https://github.com/sgl-project/sglang/pull/10158) , and new quantization adaptation refer to compressed tensor: https://github.com/vllm-project/vllm-ascend/pull/4036 .
Known issue: 
- Use export VLLM_ASCEND_ENABLE_NZ=0 to avoid error when processing weights.
- expert parallel is not supported now.
- Some models have accuracy issue, Qwen3-235B-A22B-Instruct-AWQ Qwen3-VL-235B-A22B-Instruct-AWQ and Qwen2.5-VL-32B-Instruct-AWQ. More infomation at #4378 
### How was this patch tested?

- vLLM version: v0.12.0
- vLLM main: https://github.com/vllm-project/vllm/commit/ad32e3e19ccf0526cb6744a5fed09a138a5fb2f9

### CODE REVIEW

**Comment by MengqingCao** [comment]
@paulyu12 this pr implement AWQ quantization, and now it is under testing, just at you to take a look

**Comment by menogrey** [comment]
Validation on this issue https://github.com/vllm-project/vllm-ascend/issues/4378

**Comment by wangxiyuan** [comment]
Please rebase to main now.

File: vllm_ascend/quantization/awq/awq.py [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+            return AWQMoEAscendMethod(self)
+        return None
+
+
+class AWQLinearAscendMethod(AWQLinearMethod):
+
+    def __init__(self, quant_config: AWQQuantConfig):
+        self.quant_config = quant_config
+
+    def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
```
Comment (SlightwindSec) [inline vllm_ascend/quantization/awq/awq.py:213]: **This is a solid solution for unifying weight formats.**
By preprocessing the weights during the loading phase, you effectively resolve the packing layout discrepancy between GPU and NPU, which is critical for compatibility.

**Optimization Suggestions:**
However, for larger models, the current `for` loop might introduce noticeable latency due to repeated kernel launches. Additionally, bitwise shift operations (`>>`, `<<`) can be less efficient than arithmetic operations on NPU hardware.

I suggest refactoring this implementation to **vectorize the logic** (eliminating the loop) and **replace the bitwise shifts with equivalent multiplication and integer division**. This should significantly improve the loading performance.

Comment (menogrey) [reply vllm_ascend/quantization/awq/awq.py:213]: Thanks for your suggestions. I try to remove the `for` loop but an OOM problem occurred, and the fact is if you want to execute the unpack operation, you have to alloc a big tensor (8x times memory for one tensor), and need more temporary variable if not using a in-place operation (like the bitwise operation). I think it is a trade-off between the memory usage and execution efficiency, but for a quantization, the memory usage is more important, so i won't change it, unless you have some better suggestions. @SlightwindSec

Comment (SlightwindSec) [reply vllm_ascend/quantization/awq/awq.py:213]: Fair point. I agree that avoiding OOM is the top priority here.

**Comment by ZhongsJie** [comment]
Any progress on this?

**Comment by menogrey** [comment]
> Any progress on this?

@ZhongsJie I will resume work on this PR following the completion of the quantization module refactoring.

**Comment by ZhongsJie** [comment]
> > Any progress on this?
> 
> @ZhongsJie I will resume work on this PR following the completion of the quantization module refactoring.

@menogrey I’m reworking this part in https://github.com/vllm-project/vllm-ascend/pull/7672 — if you have time, could you take a look?

**Comment by menogrey** [comment]
> > > Any progress on this?
> > 
> > 
> > @ZhongsJie I will resume work on this PR following the completion of the quantization module refactoring.
> 
> @menogrey I’m reworking this part in #7672 — if you have time, could you take a look?

That's great! In addition there are two reason that this work is pending
* I want to refactor the quantization module to align with the vllm upstream cause the upstream use better kernel abstraction. Refer to https://github.com/vllm-project/vllm-ascend/pull/6644
* The current w4a16 implement have some issues on some models as i showed above. And you should refer to the modelslim w4a16 implement.https://github.com/vllm-project/vllm-ascend/pull/4516
